### ISSUE TITLE
jpm sign: add --signing-timeout option

### ISSUE THREAD

**Issue by dstillman** [mozilla-jetpack/jpm#514]
With very large extensions, jpm sign can time out before validation completes. This option ([suggested](https://mail.mozilla.org/pipermail/dev-addons/2016-March/000450.html) by @kumar303) would allow overriding signedStatusCheckTimeout, which currently defaults to 2 minutes, at runtime.

(I assume there's a server-side timeout as well, so not sure what the upper limit is here.)

**Comment by kumar303** [comment]
We can add this to jpm but I want to clarify that if signing takes longer than two minutes there probably was a server error. In *most* cases the jpm client will be able to figure out when there is a server error but in certain cases it cannot so introducing a longer timeout wouldn't help anyone.

**Comment by dstillman** [comment]
Well, it's not the case that it wouldn't help anyone — at least with Zotero, jpm sign now regularly times out, but validation is definitely still going and eventually finishes with a successfully signed XPI that can be retrieved from the download URL in the "last status" JSON shown in the jpm error. We're fine manually changing the hard-coded timeout, but I would imagine there are at least a few other large extensions that might run into this occasionally. Possibly not enough to merit making it configurable, though, so if you want to WONTFIX this, that's fine with me.

**Comment by kumar303** [comment]
Other people have asked for this option so I think we should add it

**Comment by kumar303** [comment]
Waiting on https://github.com/mozilla-jetpack/jpm/issues/519 before adding this feature
